### PR TITLE
Recommend prefixing category methods

### LINKED ISSUE #24: Include prefix in category names and category method names on classes that aren't yours
I generally follow this rule: In a category on a class you don't own, prefix the name of the category as you would a class (in caps) in both the file name and the actual name. Prefix that category's method names with that prefix in lowercase, followed by an underscore. For example `UIFont+NYTAdditions.h`, `UIFont (NYTAdditions)`, `+ (UIFont *)nyt_titleFont`.

### PR DESCRIPTION
**Pull request by cdzombak**
closes #24

### CODE REVIEW

**Comment by cdzombak** [comment]
[pretty-printed Categories section](https://github.com/NYTimes/objective-c-style-guide/blob/f01a0c650a64f321655fc9375fd27b153695c80e/README.md#categories) as of https://github.com/NYTimes/objective-c-style-guide/commit/f01a0c650a64f321655fc9375fd27b153695c80e.

**Comment by mattbischoff** [comment]
We should do this.

File: README.md [hunk 1]
```diff
@@ -249,6 +249,24 @@ Categories may be used to concisely segment functionality and should be named to
 @interface NSString (NYTAdditions)
 ```
 
+Methods and properties added in categories should be named with an app- or organization-specific prefix. This avoids unintentionally overriding an existing method, and it reduces the chance of two categories from different libraries adding a method of the same name. (The Objective-C runtime doesn’t specify which method will be called in the latter case, which can lead to “interesting” effects.)
```
Comment (mattbischoff) [inline README.md]: Not sure about the hype after the word app.

Comment (mattbischoff) [inline README.md]: “interesting” → unintended?

**Comment by Twigz** [comment]
Merging. Prefixing category methods in Objective-C is good practice.

**Pull request merged**